contracts/strategies/StrategyBase.sol

```diff
@@ -0,0 +1,298 @@
+import {SignedDecimalMath} from "@lyrafinance/core/contracts/synthetix/SignedDecimalMath.sol";
+
+contract StrategyBase is VaultAdapter {
+  using DecimalMath for uint;
+  using SignedDecimalMath for int;
+
+  LyraVault public immutable vault;
+  OptionType public immutable optionType;
+  GWAVOracle public immutable gwavOracle;
+
+  /// @dev asset used as collateral in AMM to sell. Should be the same as vault asset
+  IERC20 public collateralAsset;
+
+  mapping(uint => uint) public lastTradeTimestamp;
+
+  uint public activeExpiry;
+  uint[] public activeStrikeIds;
+  mapping(uint => uint) public strikeToPositionId;
+
+  // basic strategy parameters
+  struct BaseStrategy {
+    uint minTimeToExpiry;
+    uint maxTimeToExpiry;
+    int targetDelta;
+    uint maxDeltaGap;
+    uint minVol;
+    uint maxVol;
+    uint size;
+    uint maxVolVariance;
+    uint gwavPeriod;
+  }
+
+  BaseStrategy public baseStrategy;
+
+  ///////////
+  // ADMIN //
+  ///////////
+
+  modifier onlyVault() virtual {
+    require(msg.sender == address(vault), "only Vault");
+    _;
+  }
+
+  constructor(
+    LyraVault _vault,
+    OptionType _optionType,
+    GWAVOracle _gwavOracle
+  ) VaultAdapter() {
+    vault = _vault;
+    optionType = _optionType;
+    gwavOracle = _gwavOracle;
+  }
+
+  function initAdapter(
+    address _curveSwap,
+    address _optionToken,
+    address _optionMarket,
+    address _liquidityPool,
+    address _shortCollateral,
+    address _synthetixAdapter,
+    address _optionPricer,
+    address _greekCache,
+    address _quoteAsset,
+    address _baseAsset,
+    address _feeCounter
+  ) external onlyOwner {
+    // set addressese for LyraVaultAdapter
+    setLyraAddresses(
+      _curveSwap,
+      _optionToken,
+      _optionMarket,
+      _liquidityPool,
+      _shortCollateral,
+      _synthetixAdapter,
+      _optionPricer,
+      _greekCache,
+      _quoteAsset,
+      _baseAsset,
+      _feeCounter
+    );
+
+    quoteAsset.approve(address(vault), type(uint).max);
+    baseAsset.approve(address(vault), type(uint).max);
+    collateralAsset = _isBaseCollat() ? baseAsset : quoteAsset;
+  }
+
+  /**
+   * @dev update the base strategy for the new round.
+   */
+  function setBaseStrategy(BaseStrategy memory _baseStrategy) external onlyOwner {
+    (, , , , , , , bool roundInProgress) = vault.vaultState();
+    require(!roundInProgress, "cannot change strategy if round is active");
+    baseStrategy = _baseStrategy;
+  }
+
+  ///////////////////
+  // VAULT ACTIONS //
+  ///////////////////
+
+  /**
+   * @dev set the board id that will be traded for the next round
+   * @param boardId lyra board Id.
+   */
+  function _setBoard(uint boardId) internal {
+    Board memory board = getBoard(boardId);
+    require(_isValidExpiry(board.expiry), "invalid board");
+    activeExpiry = board.expiry;
+  }
+
+  /**
+   * @dev exchange asset back to collateral asset and send it back to the vault
+   */
+  function _returnFundsToVaut() internal {
```

Comment:
this one may be a bit too constricting too. Some vaults may not want to exchange back to Base or Quote.

Reply:
I feel like this one is more commonly used and not just validation, how about we mark it as `virtual` so people can override?

Reply:
sure that works too